.github/workflows/reusable-mac-ci.yml

```diff
@@ -0,0 +1,170 @@
+    # *-latest images are not always the latest available versions. The minimum version of the macOS image
+    # with clang 15 is macos-13 (see
+    # https://github.com/actions/runner-images/blob/main/images/macos/macos-13-Readme.md#language-and-runtime),
+    # so we'll use that one for now, as clang 15 is the supported compiler for Fast DDS in macOS.
+    # (see https://github.com/eProsima/Fast-DDS/blob/master/PLATFORM_SUPPORT.md#compilers)
+    runs-on: macos-13
+    if: ${{ !contains(github.event.pull_request.labels.*.name, 'skip-ci') }}
+    strategy:
+      fail-fast: false
+      matrix:
+        cmake-build-type:
+          - 'RelWithDebInfo'
+    steps:
+      - name: Sync eProsima/Fast-DDS repository
+        uses: eProsima/eProsima-CI/external/checkout@feature/fastdds_mac_ci_support
+        with:
+          path: src/fastrtps
+          ref: ${{ inputs.fastdds-branch }}
+
+      - uses: eProsima/eProsima-CI/external/setup-python@feature/fastdds_mac_ci_support
+        with:
+          python-version: '3.11'
+
+      - name: Get minimum supported version of CMake
+        uses: lukka/get-cmake@latest
+        with:
+          cmakeVersion: '3.22.6'
+
+      - name: Install brew dependencies
+        uses: eProsima/eProsima-CI/macos/install_brew_packages@feature/fastdds_mac_ci_support
+        with:
+          packages: llvm asio tinyxml2 openssl@3.0
+          update: false
+          upgrade: false
+
+      - name: Install colcon
+        uses: eProsima/eProsima-CI/multiplatform/install_colcon@feature/fastdds_mac_ci_support
+
+      - name: Install Python dependencies
+        uses: eProsima/eProsima-CI/multiplatform/install_python_packages@feature/fastdds_mac_ci_support
+        with:
+          packages: vcstool xmlschema
+          upgrade: false
+
+      - name: Setup CCache
+        uses: eProsima/eProsima-CI/external/setup-ccache-action@feature/fastdds_mac_ci_support
+
+      # TODO(eduponz)
+      # - name: Update known hosts file for DNS resolver testing
+      #   if: ${{ !contains(github.event.pull_request.labels.*.name, 'no-test') }}
+      #   run: |
+      #     sudo echo ""                                                >> /etc/hosts
+      #     sudo echo "# Hosts for Fast DDS testing of DNS feature"     >> /etc/hosts
+      #     sudo echo "127.0.0.1                 localhost.test"        >> /etc/hosts
+      #     sudo echo "::1                       localhost.test"        >> /etc/hosts
+      #     sudo echo "154.56.134.194            www.eprosima.com.test" >> /etc/hosts
+      #     sudo echo "216.58.215.164            www.acme.com.test"     >> /etc/hosts
+      #     sudo echo "2a00:1450:400e:803::2004  www.acme.com.test"     >> /etc/hosts
+      #     sudo echo "140.82.121.4              www.foo.com.test"      >> /etc/hosts
+      #     sudo echo "140.82.121.3              www.foo.com.test"      >> /etc/hosts
+      #     sudo echo "ff1e::ffff:efff:1         acme.org.test"         >> /etc/hosts
```

Comment:
Shall we address this before merging?

Reply:
No. In fact, this is what blocked merging the PR two months ago. Since then we've decided that having some macOS CI on Github actions (even though it lacks some things) is better than not having it at all.